### PR TITLE
fix: check if PR is modified by verifying all commits' signatures

### PR DESCRIPTION
**Pull request by joachimroeleveld**

### CODE REVIEW

File: src/common/merge.ts [hunk 1]
```diff
@@ -20,12 +25,59 @@ export interface PullRequestDetails {
...
-  new Promise((resolve: () => void): void => {
+const getIsModified = async (
+  octokit: ReturnType<typeof getOctokit>,
+  query: {
+    pullRequestNumber: number;
+    repositoryName: string;
+    repositoryOwner: string;
+  },
+): Promise<boolean> => {
+  const iterator = makeGraphqlIterator<
```
Comment (joachimroeleveld) [inline src/common/merge.ts:36]: Using an iterator here because the the list should be considered infinite and it's more efficient than a recursive function.

File: src/common/makeGraphqlIterator.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+/* eslint-disable require-atomic-updates */
+/* eslint-disable prefer-destructuring */
+/* eslint-disable immutable/no-let */
```
Comment (merlinnot) [inline src/common/makeGraphqlIterator.ts:3]: Sort 😉

File: src/common/makeGraphqlIterator.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  edges: Array<{
+    node: Iterable;
+  }>;
+  pageInfo: {
+    endCursor: string;
+    hasNextPage: boolean;
+  };
+}
+
+export const makeGraphqlIterator = <ResponseData, IterableData>(
```
Comment (merlinnot) [inline src/common/makeGraphqlIterator.ts:19]: What was the reason to choose manual creation based on symbols instead of actual `export const makeGraphqlIterator = async function*`?

File: src/common/merge.ts [hunk 4]
```diff
@@ -20,6 +25,54 @@ export interface PullRequestDetails {
...
+    findPullRequestCommits,
+    query,
+    (
+      response: FindPullRequestCommitsResponse,
+    ): IterableList<PullRequestCommitNode> =>
+      response.repository.pullRequest.commits,
+  );
+
+  // eslint-disable-next-line immutable/no-let
+  let originalAuthor: string | undefined = undefined;
```
Comment (merlinnot) [inline src/common/merge.ts:50]: Seems like we can use the `next` method for direct assignment here, avoiding the need to `let`?

File: src/common/makeGraphqlIterator.ts (hunk 3 above)
Comment (joachimroeleveld) [reply src/common/makeGraphqlIterator.ts]: Ah, that's been a long time. Forgot that it existed. Changed :).

File: src/eventHandlers/continuousIntegrationEnd/index.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 /* eslint-disable no-await-in-loop */
 
 import { context, getOctokit } from '@actions/github';
-import type { GraphQlQueryResponseData } from '@octokit/graphql';
+import { GraphQlQueryResponseData } from '@octokit/graphql';
```
Comment (merlinnot) [inline src/eventHandlers/continuousIntegrationEnd/index.ts:4]: Why the change?

Comment (joachimroeleveld) [reply src/eventHandlers/continuousIntegrationEnd/index.ts]: I think I used it before, not anymore. Fixed.

File: src/common/merge.ts [hunk 6]
```diff
@@ -143,8 +194,12 @@ export const tryMerge = async (
...
     logInfo(`Pull request version bump is not allowed by PRESET.`);
   } else if (
-    isMatch(commitAuthorName, allowedAuthorName) === false &&
-    disabledForManualChanges === true
+    enabledForManualChanges === false &&
+    (await getIsModified(octokit, {
+      pullRequestNumber,
+      repositoryName,
+      repositoryOwner,
+    })) === true
```
Comment (merlinnot) [inline src/common/merge.ts:202]: Do you think we could assign it to a constant instead of inlining to improve readability?

**Review by merlinnot — APPROVED** [review]

File: src/common/merge.ts (hunk 6 above)
Comment (joachimroeleveld) [reply src/common/merge.ts:202]: We can, but it will make execution less efficient. Wdyt?

Comment (merlinnot) [reply src/common/merge.ts:202]: We can still evaluate it lazily if it's in the else block. I didn't mean to put it before this chain.
